src/js/client/editprefset-viewport.js

```diff
@@ -0,0 +1,930 @@
+/**
+ * Edit Preferences Viewport
+ *
+ * This is the primary entry point and container for the full page
+ * preference safe editor. The grade `gpii.devpmt.editPrefs` orchestrates
+ * the entire editor including it's model and subcomponents.
+ *
+ * Copyright 2019 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+/* global saveAs, Foundation */
+"use strict";
+
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+
+/**
+ * A devpmt tailored view component that brings in some common
+ * traits for our rendered widgets.
+ *
+ * Builds off of `gpii.handlbars.templateAware` and `gpii.binder.bindOnCreate`
+ * grades.  Includes a standard renderInitialMarkup invoker that uses
+ * the defined initial template and the model as the prefsSet for rendering.
+ * Also includes a useful reRender invoker that can be bound to model
+ * listeners or invoked programmatically.
+ */
+fluid.defaults("gpii.devpmt.viewComponent", {
+    gradeNames: ["gpii.handlebars.templateAware.serverMessageAware", "gpii.binder.bindOnCreate",
+        "gpii.binder.bindMarkupEvents"],
+    invokers: {
+        renderInitialMarkup: {
+            func: "{that}.renderMarkup",
+            args: ["initial", "{that}.options.templates.initial", "{that}.model"]
+        },
+        reRender: {
+            func: "{that}.events.refresh.fire"
+        }
+    }
+});
+
+/**
+ * Edit Prefs is the main component for driving the page editing
+ * a preference set.
+ */
+fluid.defaults("gpii.devpmt.editPrefs", {
+    gradeNames: ["gpii.devpmt.viewComponent"],
+    mergePolicy: {
+        allSolutions: "noexpand",
```

Comment:
This is plenty wierd - how do values get into these options, and why mustn't they be expanded?

Reply:
They are added by the gpii-handlebars helper tag that creates the component on the page template.  They are listed as "noexpand", because some of the entries and syntax in them are interpreted as IoC references, variables, or other things during the component installation which tries to mangle them into something else.

Granted, there are a few other things that need to be included on a round of a ResourceLoader-isation, but I'd like to jira that and make it part of the next round of work.